impress-org/givewp-next-gen
# Feature: add separate route for form preview and introduce `FormSettings` property

## Description



This PR separates the donation form's preview route and render route.  Essentially this means the preview route will accept params that can override the form's rendering logic for previewing changes, while the render route is confined to _only_  what's stored in the database using donation form ID.

Along with the new route, the Donation Form's `settings` property was updated with a new `FormSettings` property.  Everywhere that property was previously being used as an array has now been updated to the new api.

## Affects



The form builder preview route and form settings property.

## Visuals



N/A

## Testing Instructions



Just make sure the preview, settings, and block rendering works as expected.  There should be no visual change or issues.

## Pre-review Checklist
